ihttp/ihttp.go

```diff
@@ -97,6 +99,18 @@ func (s Handler) Handle(w http.ResponseWriter, req *http.Request) {
 
 		return
 	}
+
+	if len(s.Patch) > len(binary.MagicString) {
+		log.Info("patch string is too long, ignoring")
+	} else if len(s.Patch) > 0 {
+		index := bytes.Index(file, binary.MagicString)
+		if index >= 0 {
+			copy(file[index:], s.Patch)
```

Comment:
The copy will overwrite the iPXE contents indefinitely (or at least until another patch is applied). I.e. all subsequent requests will receive the patch which seems undesirable.

Was this intentional or are you trying to do per-request patching?

Reply:
Not intended. I'll fix it to make a copy before patching.